src/app/network/upload.test.ts

```diff
@@ -0,0 +1,57 @@
+import { describe, it, expect, vi, beforeEach } from 'vitest';
+import { uploadFile } from './upload';
+import { ConnectionLostError } from './requests';
+
+const uploadMock = vi.fn();
+
+const uploadParams = {
+  filesize: 50000,
+  filecontent: new File(['content'], 'file.txt'),
+  creds: { user: 'user', pass: 'password' },
+  mnemonic: 'mnemonic123',
+  progressCallback: vi.fn(),
+};
+
+vi.mock('./NetworkFacade', () => ({
+  NetworkFacade: vi.fn().mockImplementation(() => ({
+    uploadMultipart: vi.fn().mockResolvedValue('multipart-upload-success'),
+    upload: uploadMock,
+  })),
+}));
+
+describe('uploadFile', () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it('should successfully upload a file', async () => {
+    uploadMock.mockResolvedValue('upload-success');
+
+    const result = await uploadFile('bucket123', uploadParams);
+
+    expect(result).toBe('upload-success');
+  });
+
+  it('should retry upload on failure and succeed on second attempt', async () => {
```

Comment:
You have to assert that the function inside `uploadFile` that you expect to be called again is, in fact, called again the same times that you are expecting it to be called (in this case, two).